lib/cfg-parser.c

```diff
@@ -462,23 +462,20 @@ cfg_parser_cleanup(CfgParser *self, gpointer instance)
 gboolean
 cfg_process_flag(CfgFlagHandler *handlers, gpointer base, const gchar *flag_)
 {
-  gint h;
-  gchar flag[32];
+  gsize h;
+  gboolean result = FALSE;
 
-  for (h = 0; flag_[h] && h < sizeof(flag); h++)
+  GString *flag = g_string_new(flag_);
+  for (h = 0; flag->str[h] && h < flag->len; h++)
     {
-      if (flag_[h] == '_')
-        flag[h] = '-';
-      else
-        flag[h] = flag_[h];
+      if (flag->str[h] == '_')
+        flag->str[h] = '-';
     }
-  flag[h] = 0;
```

Comment:
we could also use normalize_flag() here.

Reply:
Thanks, with that the function becomes much simpler!